laterpay/application/Helper/Pricing.php

```diff
@@ -41,19 +41,38 @@ public static function is_purchasable( $post_id = null ) {
         // check, if the current post price is not 0
         $price = LaterPay_Helper_Pricing::get_post_price( $post_id, true );
 
-        $only_time_passes_allowed = get_option( 'laterpay_only_time_pass_purchases_allowed' );
+        // Get current post price behaviour.
+        $post_price_behaviour = (int) get_option( 'laterpay_post_price_behaviour' );
 
         // Getting list of timepass by post id.
         $time_passes_list = LaterPay_Helper_TimePass::get_time_passes_list_by_post_id( $post_id, null, true );
 
         // Getting list of subscription by post id.
         $subscriptions_list = LaterPay_Helper_Subscription::get_subscriptions_list_by_post_id( $post_id, null, true );
 
-        if ( ( ( floatval( 0.00 ) === floatval( $price ) || $only_time_passes_allowed )&& ( ( 0 === count( $time_passes_list ) ) && ( 0 === count( $subscriptions_list ) ) ) ) || ! in_array( get_post_type( $post_id ), (array) get_option( 'laterpay_enabled_post_types' ), true ) ) {
+        // Global Price Value.
+        $global_default_price = get_option( 'laterpay_global_price' );
+
+        if ( ( ( floatval( 0.00 ) === floatval( $price ) || 1 === $post_price_behaviour || ( 2 === $post_price_behaviour && floatval( 0.00 ) === (float) $global_default_price ) ) && ( ( 0 === count( $time_passes_list ) ) && ( 0 === count( $subscriptions_list ) ) ) ) || ! in_array( get_post_type( $post_id ), (array) get_option( 'laterpay_enabled_post_types' ), true ) ) {
```

Comment:
Should we make this more readable? Can break it in multiple variables and use it in if like below. 
```php 
$post_type_enabled = ( ! in_array( get_post_type( $post_id ), (array) get_option( 'laterpay_enabled_post_types' ), true ) );
$main_condition_name = (
                $post_type_enabled
                ||
                $condition_2_name
                ...
);
if ( $main_condition_name )
```

This can be used at multiple places in this PR.